python/foglamp/tasks/backup_restore/backup.py

```diff
@@ -0,0 +1,667 @@
+                    or for a specific state only
+        Returns:
+        Raises:
+        """
+
+        pass
+
+
+class BackupProcess(FoglampProcess):
+    """ Backups the entire FogLAMP repository into a file in the local filesystem,
+        it executes a full warm backup
+    """
+
+    _MODULE_NAME = "foglamp_backup"
+
+    _MESSAGES_LIST = {
+
+        # Information messages
+        "i000001": "Execution started.",
+        "i000002": "Execution completed.",
+
+        # Warning / Error messages
+        "e000000": "general error",
+        "e000001": "cannot initialize the logger - error details |{0}|",
+        "e000002": "cannot retrieve the configuration from the manager, trying retrieving from file "
+                   "- error details |{0}|",
+        "e000003": "cannot retrieve the configuration from file - error details |{0}|",
+        "e000004": "cannot delete/purge backup file on file system - id |{0}| - file name |{1}| error details |{2}|",
+        "e000005": "cannot delete/purge backup information on the storage layer "
+                   "- id |{0}| - file name |{1}| error details |{2}|",
+        "e000007": "backup failed.",
+        "e000008": "cannot execute the backup, either a backup or a restore is already running - pid |{0}|",
+        "e000009": "cannot retrieve information for the backup id |{0}|",
+        "e000010": "directory used to store backups doesn't exist - dir |{0}|",
+        "e000011": "directory used to store semaphores for backup/restore synchronization doesn't exist - dir |{0}|",
+        "e000012": "cannot create the configuration cache file, neither FOGLAMP_DATA or FOGLAMP_ROOT are defined.",
+        "e000013": "cannot create the configuration cache file, provided path is not a directory - dir |{0}|",
+    }
+    """ Messages used for Information, Warning and Error notice """
+
+    _CONFIG_FILE = "configuration_cache.json"
+    """ Stores a configuration cache in case the configuration Manager is not available"""
+
+    # Configuration retrieved from the Configuration Manager
+    _CONFIG_CATEGORY_NAME = 'BACK_REST'
+    _CONFIG_CATEGORY_DESCRIPTION = 'Configuration for backup and restore operations'
+
+    _CONFIG_DEFAULT = {
+        "host": {
+            "description": "Host server for backup and restore operations.",
+            "type": "string",
+            "default": "localhost"
+        },
+        "port": {
+            "description": "PostgreSQL port for backup and restore operations.",
+            "type": "integer",
+            "default": "5432"
+        },
+        "database": {
+            "description": "Database to manage for backup and restore operations.",
+            "type": "string",
+            "default": "foglamp"
+        },
+        # FIXME:
+        "backup_dir": {
+            "description": "Directory where the backups will be created.",
+            "type": "string",
+            "default": "/tmp"
+        },
+        # FIXME:
```

Comment:
@izoratti please define default

Reply:
defines (with Ivan) details in : https://scaledb.atlassian.net/browse/FOGL-746